src/components/input-field/input-field.tsx

```diff
@@ -391,6 +393,8 @@ export class InputField {
 
         if (this.invalid) {
             this.mdcTextField.valid = false;
+        } else {
+            this.mdcTextField.valid = true;
```

Comment:
```suggestion
        this.mdcTextField.valid = !this.invalid;
```

Reply:
Just because invalid isn't set to true doesn't mean the field should be invalid.

If there's a need to clear the invalid state, we need to be sure that's indeed the intention, otherwise we break existing behaviour.